editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -1035,6 +1035,29 @@ void CanvasItemEditor::_viewport_input_event(const InputEvent &p_event) {
 
 		const InputEventMouseButton &b = p_event.mouse_button;
 
+		if (b.button_index == GESTURE_MAGNIFY) {
+
+			if (zoom < MIN_ZOOM) {
```

Comment:
Yep it's totally worth to make a zoom function and replace all the duplicated lines. And your idea about the button wheel events are great, which is also worth implement :D